app/uk/gov/hmrc/agentsubscriptionfrontend/models/AddressValidator.scala

```diff
@@ -28,74 +28,83 @@ import scala.util.matching.Regex
 
 object AddressValidator {
 
-  type ValidationErrors = NonEmptyList[ValidationError]
+  type ValidatedType[A] = ValidatedNel[ValidationError, A]
```

Comment:
We can either define validatedType for each type,
val validatedDesAddress = ValidatedNel[ValidationError, DesAddress]
val validatedLine = ValidatedNel[ValidationError, String]
etc...
or this generic validatedType[A]